### ISSUE TITLE
Content directory permissions

### ISSUE THREAD

**Issue by Lezek123** [Joystream/joystream#2911]
## Current (Giza) permissions model
See: https://github.com/Joystream/joystream/blob/giza/runtime-modules/content/src/permissions/mod.rs#L7

## The issue
Noone, except member's channel owner and collaborators (if any), is able to remove channel/video assets and the channel/video itself.
Because storage and distributor nodes by default follow their obligations assigned by the chain, this means sensitive content inside member's channel can only really be removed from the platform by:
- `sudo` (using `sudoAs`)
- through an off-chain consensus of all storage providers.
- through the removal of storage obligations on the storage bag level by the storage working group lead (in that case the data objects will not be stored/served, but they will remain in the runtime state)

### Change permissions model to allow Curators and/or Lead to remove channel/video assets

This is a very simple fix, as it only requires a small change in [`ensure_actor_authorized_to_update_channel_assets`](https://github.com/Joystream/joystream/blob/giza/runtime-modules/content/src/permissions/mod.rs#L241) function and a corresponding change in the Joystream CLI.

#### Attack vector

However this has a major drawback, because it means the entire content directory can be wiped out by a single actor.
Another issue is that since there is a `deletion_prize` associated with removing data objects from the storage, and the prize currently goes to [whoever triggered the deletion](https://github.com/Joystream/joystream/blob/giza/runtime-modules/content/src/lib.rs#L805), such an attack (wiping out of the content directory) may even become profitable in some cases (especially in Giza, since there are no transaction fees)

**Ways to mitigate:**
- ensure that removing assets by curator/lead is costly enough that such an attack becomes unlikely
- on the storage node level: do not follow data object deletion immediately, but make it a delayed process instead (in Giza this doesn't apply because auto-deletion is disabled). This at least prevents the system from losing the data objects themselves, but would still require fixing the runtime state in case the attack occurs.

 ### Change permissions model to allow Curators and/or Lead to manage channels/videos that have been censored for at least `x` blocks

This solution is slightly more complex, but much more secure.
The idea is to make the channel/video deletion a 2-step process. In the first step the lead/curator censors the sensitive content. This can be reflected by storage/distributor nodes not serving the content, but keeping it stored in case it is uncensored later.
Only after the content remains censored for at least `x` blocks, giving other curators/lead a reasonable amount of time to react in case the action was performed by a bad actor, the content can be removed by a lead/curator. This should completely eliminate the risk of entire content directory beeing wiped out by a bad actor, without sacrificing the ability for curators to quickly react and shut down any sensitive content.
